executor/builder.go

```diff
@@ -1025,13 +1030,17 @@ func (b *executorBuilder) buildHashJoin(v *plannercore.PhysicalHashJoin) Executo
 		e.outerFilter = v.LeftConditions
 		e.innerKeys = v.RightJoinKeys
 		e.outerKeys = v.LeftJoinKeys
-		if defaultValues == nil {
+	}
+	if defaultValues == nil {
+		if e.outerHashJoin {
```

Comment:
This is not correct here.
defaultValues is used to fill the null-vals for outerJoin.
The count of defaultValues is always equal to the `e.innerExec.Schema().Len()`

Reply:
When defaultValues == nil, it should be initialized to be null-vals according to the coorresponding  inner side. This follows the logic of the former codes. Here seems ok.

Reply:
> This is not correct here.
> defaultValues is used to fill the null-vals for outerJoin.
> The count of defaultValues is always equal to the `e.innerExec.Schema().Len()`

When defaultValues == nil, it should be initialized to be null-vals according to the coorresponding inner side. This follows the logic of the former codes. Here seems ok.